leapp/utils/report.py

```diff
@@ -1,72 +1,24 @@
 import json
-import six
-
-from jinja2 import Template
 
 from leapp.utils.audit import get_messages
 
 
-def _flatten_dict(dict_):
-    new_dict = {}
-    if type(dict_) is not dict:
-        return new_dict
-
-    for k, v in six.iteritems(dict_):
-        if type(v) == dict:
-            new_dict.update(_flatten_dict(v))
-        else:
-            new_dict[k] = v
-
-    return new_dict
-
-
-def fetch_upgrade_report_raw(context_id, renderers=True):
+def fetch_upgrade_report_messages(context_id):
     """
     :param context_id: ID to identify the needed messages
     :type context_id: str
-    :param renderers: whether copy or skip renderers field of original message
-    :type boolean
-    :return: A list of upgrade messages of type "Report" withing the given context,
-             each message is a dict of raw data.
+    :param renderer: How messages will be rendered (e.g html or plaintext)
+    :type renderer: str
+    :return: All upgrade messages of type "Report" withing the given context
     """
     report_msgs = get_messages(names=['Report'], context=context_id) or []
+    if not report_msgs:
```

Comment:
That's not needed iiuc, report_msgs will be an empty list and cycle at L19-L22 will be skipped

Reply:
fixed